ui/analyse/src/keyboard.ts

```diff
@@ -116,19 +114,12 @@ export const bind = (ctrl: AnalyseCtrl) => {
 };
 
 export function view(ctrl: AnalyseCtrl): VNode {
-  return snabModal({
-    class: 'keyboard-help',
-    onInsert: async ($wrap: Cash) => {
-      const [, html] = await Promise.all([
-        lichess.loadCssPath('analyse.keyboard'),
-        xhr.text(xhr.url('/analysis/help', { study: !!ctrl.study })),
-      ]);
-      $wrap.find('.scrollable').html(html);
-    },
+  return snabDialog({
+    class: 'help.keyboard-help',
+    htmlUrl: '/analysis/help',
     onClose() {
       ctrl.keyboardHelp = false;
       ctrl.redraw();
     },
-    content: [h('div.scrollable', spinner())],
   });
```

Comment:
Yeah, I toyed with a spinner mechanism in the dialog itself while content was loading. I removed it to streamline things. I figured half a second was OK to wait on a dialog to appear. But overall it's maybe 20 lines due to some messiness it introduces to the various paths to construct the dialogs. It can be default, but probably a conditional in opts in case delayed visibility is preferred. Want we should add that?

Reply:
Another approach, if it's a show now modal dialog, would be to pre-darken everything to $page-mask and superimpose the spinner directly on the UI until the layout arrives.